coderd/httpmw/logger.go

```diff
@@ -74,3 +81,37 @@ func Logger(log slog.Logger) func(next http.Handler) http.Handler {
 		})
 	}
 }
+
+type RequestLoggerContext struct {
+	Fields map[string]any
+
+	log     *slog.Logger
+	written bool
+}
+
+func (c *RequestLoggerContext) WriteLog(ctx context.Context, msg string, status int) {
```

Comment:
These logs should look like the existing logs. We currently set the message to `r.Method` and add status as a field.

Reply:
Since `r.Method` will be known when the `RequestLoggerContext` is created, it should probably just be stored on the struct and not included as a parameter here.